### DISCUSSION TITLE
Variables and Constraints passing

### DISCUSSION THREAD

**Discussion by tianmeioop** [yalmip/YALMIP#1388] [General discussion]
I have an optimization model as shown below：
```
function f=u1(C,y)
x=sdpvar(1,1);
f=x+y;
C=C+[x<=10];
ops=sdpsettings('solver','gurobi');
result=optimize(C,-f,ops);
```
As defined by the function input parameter, I intend to add constraint C and variable y to the optimization model.
For example, now add the following constraints and variables:
```
y=sdpvar(1,1);
f=u1([y<=10],y);
```
Reconstruct the optimization model in this way to see if it is in line with the use of yalmip？
Because I've been doing two-stage robust optimization lately, using the C&CG algorithm, I add variables and constraints to the main problem. So there is such a problem to deal with.

**Comment by johanlofberg** [comment]
Sure, but x won't be easily accessed outside the function, and I doubt you want to return the objective function x+y from the function. feels more typical that you would want to return the optimal objective value after solving the sub-problem

**Reply by tianmeioop** [reply]
Thanks for your answer!
